Samples - Switched the build_component sample to the new container API

Ark-kun opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/kubeflow/pipelines/2279)
<!-- Reviewable:end -->

Ark-kun commented:
@hongye-sun Do you have any objection to this change? It removes quite a lot of boilerplate and the users no longer needs to learn complicated concepts like compilation.

gaoning777 commented:
what is the plan for the component build APIs? will build_python_component be deprecated? If not, could you create another sample to demonstrate the build_image_from_working_dir. If so, lgtm.

numerology commented:
/lgtm
You might want to wait for hearing back from @hongye-sun

Ark-kun commented:
> what is the plan for the component build APIs? will build_python_component be deprecated?

I did not initiate this deprecation. AFAIK, @SinaChavoshi has requested for the component creation APIs to be reduced to one. Same with container-building APIs. Based on this request our team had two design reviews where we ultimately agreed to merge the `build_python_component` and `func_to_container_op` and also make it easy to automatically build container from working directory when components are being created (designed by @hongye-sun and me).

Ark-kun commented:
> You might want to wait for hearing back from @hongye-sun

Yes, I'll be waiting for @hongye-sun

SinaChavoshi commented:
> > what is the plan for the component build APIs? will build_python_component be deprecated?
Just to confirm I strongly requested / recommend reducing the number of ways that users can build components / containers. I leave the decision of which path we should go with to the team however I think we should choose one way and deprecate the rest.

Ark-kun commented:
Talked with @hongye-sun offline.
He suggested:
* Pre-create the `requirements.txt` file instead of creating it dynamically in a temporary directory.
* Showcase setting the `default_base_image_or_builder` to `build_image_from_working_dir` so that it's used every time the user creates component with `func_to_container_op`.

I've implemented both of those suggestions (and also added the customized builder scenario).

Ark-kun commented:
/approve

SinaChavoshi commented on samples/core/component_build/component_build.ipynb:
```diff
@@ -88,178 +81,79 @@
-   "source": [
-    "## Create a python function"
+    "from pathlib import Path\n",
+    "print(Path('requirements.txt').read_text())"
```
Would it make sense to have this as mark up rather than code? execution of this line will case an error in the notebook as the env may not have the requirements.txt already created in it. 

Alternatively you could create the requiremetns.txt first.

Ark-kun replied:
The commit that introduced this line includes the `requirements.txt` which resides in the same directory as the notebook.

SinaChavoshi commented:
/lgtm
